pytransform3d/trajectories.py

```diff
@@ -436,6 +515,322 @@ def pqs_from_dual_quaternions(dqs):
     return out
 
 
+
+def norm_axis_angles(a):
+    """Normalize axis-angle representation.
+
+    Parameters
+    ----------
+    a : array-like, shape (..., 4)
+        Axis of rotation and rotation angle: (x, y, z, angle)
+
+    Returns
+    -------
+    a : array, shape (..., 4)
+        Axis of rotation and rotation angle: (x, y, z, angle). The length
+        of the axis vector is 1 and the angle is in [0, pi). No rotation
+        is represented by [1, 0, 0, 0].
+    """
+    angles = a[..., 3]
+    norm = np.linalg.norm(a[..., :3], axis=1)
+
+
+    res = np.ones_like(a)
+
+    # Create masks for elements where angle or norm is zero
+    zero_mask = (angles == 0.0) | (norm == 0.0)
+
+
+    non_zero_mask = ~zero_mask
+    res[non_zero_mask, :3] = (
+        a[non_zero_mask, :3] / norm[non_zero_mask, np.newaxis]
```

Comment:
see @kopytjuk 's comment: is np.newaxis really required?

Reply:
yes as norm is a 1D array and a has size (N, 3) we need to transform norm to a (N, 1) to get it correctly broadcastet and we want to perform element wise division here.